rust/automerge-c/src/doc/mark.rs

```diff
@@ -0,0 +1,294 @@
+use automerge as am;
+
+use std::any::type_name;
+
+use am::marks::{ExpandMark, Mark};
+use am::transaction::Transactable;
+use am::ReadDoc;
+
+use crate::{
+    byte_span::{to_str, AMbyteSpan},
+    item::{AMitem, Item},
+    items::AMitems,
+    obj::{to_obj_id, AMobjId},
+    result::{to_result, AMresult},
+};
+
+use super::{
+    utils::{to_doc, to_doc_mut},
+    AMdoc,
+};
+
+macro_rules! to_expand_mark {
+    ($mark_expand:expr) => {{
+        let result: Result<ExpandMark, am::AutomergeError> = (&$mark_expand).try_into();
+        match result {
+            Ok(expand_mark) => expand_mark,
+            Err(e) => return AMresult::error(&e.to_string()).into(),
+        }
+    }};
+}
+
+pub(crate) use to_expand_mark;
+
+/// \ingroup enumerations
+/// \enum AMmarkExpand
+/// \installed_headerfile
+/// \brief A mark's expansion mode for when bordering text is inserted.
+#[derive(PartialEq, Eq)]
+#[repr(u8)]
+pub enum AMmarkExpand {
+    /// Include text inserted at the end offset.
+    After = 3,
+    /// Include text inserted at the start offset.
+    Before = 2,
+    /// Include text inserted at either offset.
+    Both = 4,
+    /// The default tag, not a mark expansion mode signifier.
+    Default = 0,
+    /// Exclude text inserted at either offset.
+    None = 1,
+}
```

Comment:
@ConradIrwin, if the tags' ordinals aren't meaningful then this declaration can be simplified a bit e.g.
```suggestion
pub enum AMmarkExpand {
    /// Include text inserted at the end offset.
    After = 1,
    /// Include text inserted at the start offset.
    Before,
    /// Include text inserted at either offset.
    Both,
    /// The default tag, not a mark expansion mode signifier.
    Default = 0,
    /// Exclude text inserted at either offset.
    None = 4,
}
```